### PR TITLE
Fix broadcast

### PR DESCRIPTION
**Pull request by Dewei-Wang-sh**
relax upstream broadcast constraint

### CODE REVIEW

**Comment by jopperm** [comment]
What's the difference between the upstream broadcast op and ours?

**Comment by Dewei-Wang-sh** [comment]
> What's the difference between the upstream broadcast op and ours?

we do not enforce same encoding between source and result.

File: third_party/intel/lib/TritonIntelGPUTransforms/DistributeToWarps.cpp [hunk 1]
```diff
@@ -428,7 +428,8 @@ class TritonIntelGPUDistributeToWarpsPass
         else if (auto makeRange = dyn_cast<tt::MakeRangeOp>(op))
           distributeMakeRangeOp(makeRange, warpId);
         else if (isa<tt::LoadOp, tt::DotOp, tt::AdvanceOp, tt::ReduceOp,
-                     tt::SplatOp, tt::BroadcastOp, tt::ExpandDimsOp>(op) ||
+                     tt::SplatOp, tt::BroadcastOp, ttgi::BroadcastOp,
```
Comment (jopperm) [inline third_party/intel/lib/TritonIntelGPUTransforms/DistributeToWarps.cpp:431]: Can both broadcast ops still occur?

File: third_party/intel/lib/TritonToTritonGPUWarp/TritonToTritonGPUWarpPass.cpp [hunk 2]
```diff
@@ -751,6 +766,8 @@ class ConvertTritonToTritonGPUWarp
       } else if (isa<tt::SplatOp, tt::BroadcastOp, tt::ReduceOp,
                      tt::MakeRangeOp>(def)) {
         chainedVals.insert(def->getResult(0));
+      } else if (isa<tt::ExpandDimsOp>(def)) {
+        ;
```
Comment (jopperm) [inline third_party/intel/lib/TritonToTritonGPUWarp/TritonToTritonGPUWarpPass.cpp:770]: Maybe add a comment why this is a no-op?

**Review by jopperm — APPROVED** [review]
> > What's the difference between the upstream broadcast op and ours?
> 
> we do not enforce same encoding between source and result.

Ah great, I ran into this earlier!

File: third_party/intel/lib/TritonToTritonGPUWarp/TritonToTritonGPUWarpPass.cpp (hunk 2 above)
Comment (Dewei-Wang-sh) [reply third_party/intel/lib/TritonToTritonGPUWarp/TritonToTritonGPUWarpPass.cpp:770]: sure, let me clean my thoughts

**Pull request merged**